process/peer/process.go

```diff
@@ -0,0 +1,288 @@
+package peer
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+// ShardedPeerAdapters represents a list of peer account adapters organized by the shard id
+//  each validator account is alocated to
+type ShardedPeerAdapters map[uint32]state.AccountsAdapter
+
+type peerProcess struct {
+	marshalizer marshal.Marshalizer
+	shardHeaderStorage storage.Storer
+	nodesCoordinator sharding.NodesCoordinator
+	shardCoordinator sharding.Coordinator
+	adrConv state.AddressConverter
+	peerAdapters ShardedPeerAdapters
+	mutPeerAdapters *sync.RWMutex
+}
+
+// NewPeerProcessor instantiates a new peerProcess structure responsible of keeping account of
+//  each validator actions in the consensus process
+func NewPeerProcessor(
+	peerAdapters ShardedPeerAdapters,
+	adrConv state.AddressConverter,
+	nodesCoordinator sharding.NodesCoordinator,
+	shardCoordinator sharding.Coordinator,
+	shardHeaderStorage storage.Storer,
+) (*peerProcess, error) {
+	if peerAdapters == nil {
+		return nil, process.ErrNilPeerAccountsAdapter
+	}
+	if adrConv == nil {
+		return nil, process.ErrNilAddressConverter
+	}
+	if nodesCoordinator == nil {
+		return nil, process.ErrNilNodesCoordinator
+	}
+	if shardCoordinator == nil {
+		return nil, process.ErrNilShardCoordinator
+	}
+	if shardHeaderStorage == nil {
+		return nil, process.ErrNilShardHeaderStorage
+	}
+
+	return &peerProcess{
+		mutPeerAdapters: &sync.RWMutex{},
+		peerAdapters: peerAdapters,
+		adrConv: adrConv,
+		nodesCoordinator: nodesCoordinator,
+		shardCoordinator: shardCoordinator,
+		shardHeaderStorage: shardHeaderStorage,
+	}, nil
+}
+
+// LoadInitialState takes an initial peer list, validates it and sets up the initial state for each of the peers
+func (p *peerProcess) LoadInitialState(in []*sharding.InitialNode) error {
+	for _, node := range in {
```

Comment:
initial state should be done at constructor

Reply:
Called this function in the constructor and left it for future use cases